developer-docs-site/static/messageboard_demo/client/README.md

```diff
@@ -0,0 +1,187 @@
+    account: signer, board_addr: address, message: vector<u8>
+) acquires MessageChangeEventHandle {
+    let event_handle = borrow_global_mut<MessageChangeEventHandle>(board_addr);
+    Event::emit_event<MessageChangeEvent>(
+        &mut event_handle.change_events,
+        MessageChangeEvent{
+            message,
+            participant: Signer::address_of(&account)
+        }
+    );
+}
+```
+
+### ACL-Based Messageboard
+
+[ACL-Based messageboard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/messageboard/sources/ACLMessageBoard.move) is similar to the capability based messageboard. The main difference is it uses ACL to control the access, which is essentially a list of the addresses. 
+
+To add a new participant, the board creator adds the participant into the ACL list under his account. 
+
+```tsx
+public(script) fun add_participant(account: &signer, participant: address) acquires ACLBasedMB {
+    let board = borrow_global_mut<ACLBasedMB>(Signer::address_of(account));
+    ACL::add(&mut board.participants, participant);
+}
+```
+
+When a participant want to change the pinned message, the participant first need to borrow the board resource under creator’s account and check if the participant is in the ACL. Then it borrows mutable board to modify the pinned_post. 
+
+```tsx
+public(script) fun send_pinned_message(
+    account: &signer, board_addr: address, message: vector<u8>
+) acquires ACLBasedMB, MessageChangeEventHandle {
+    let board = borrow_global<ACLBasedMB>(board_addr);
+    assert!(ACL::contains(&board.participants, Signer::address_of(account)),EACCOUNT_NOT_IN_ACL);
+
+    let board = borrow_global_mut<ACLBasedMB>(board_addr);
+    board.pinned_post = message;
+```
+
+## Interacting With Messageboard
+
+### **Compile and Publish the Modules**
+
+The [following code](https://github.com/aptos-labs/aptos-core/blob/main/developer-docs-site/static/messageboard_demo/client/index.ts) basically runs a command to compile the messageboard contract with the account address of developer. 
+
+It then read the compiled modules in mv files in the module path and creates a txn to publish the modules under the developer’s account address.
+
+```tsx
+chdir('../../../../aptos-move/move-examples/messageboard');
+execSync(
+    `aptos move compile --package-dir . --named-addresses MessageBoard=${messageboardDev.address().toString()}`
+);
+var module_path = "build/MessageBoard";
+MessageboardUtil.installMessageboard(client, messageboardDev, module_path)
+```
+
+### **Interact with the Messageboard**
+
+We use [Aptos SDK](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) to interact with Aptos Chain. 
+
+After the modules are published under the developer’s account address, we can use the functions of these published modules to specify the function name, argument to create the messageboard initialization payload and send the txn to devnet to create the messageboard 
```

Comment:
After the modules are published under the developer’s account address, we can use the functions of these published modules to specify the function name and argument to create the messageboard initialization payload, and send the transaction to devnet to create the messageboard.